app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -614,34 +522,177 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
         }
     }
 
-    private IntentWidget getPendingIntentWidget() {
-        IntentWidget bestMatch = null;
+    /**
+     * Performs any necessary relocating and scaling of an image coming from either a
+     * SignatureWidget or ImageWidget (capture or choose)
+     *
+     * @param originalImage the image file returned by the image capture or chooser intent
+     * @param shouldScale if false, indicates that the image is from a signature capture, so should
+     *                not attempt to scale
+     * @return the image file that should be displayed on the device screen when this question
+     * widget is in view
+     */
+    private File moveAndScaleImage(File originalImage, boolean shouldScale) throws IOException {
+        // We want to save our final image file in the instance folder for this form, so that it
+        // gets sent to HQ with the form
+        String instanceFolder =
+                mInstancePath.substring(0, mInstancePath.lastIndexOf("/") + 1);
+        String imageFilename = System.currentTimeMillis() + ".jpg";
```

Comment:
Actually, there are places in the comments where it talks specifically about jpg format. Are we sure there's no reason that we seem to have been insisting on that?